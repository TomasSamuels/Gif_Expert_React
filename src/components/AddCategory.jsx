import { useState } from 'react';

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChanged = ({ target }) =>{
        setInputValue( target.value );
    };

    const onSubmit = ( event ) => {
        event.preventDefault();
        const value = inputValue.trim();
        if( value.length <= 1 ) return;
        onNewCategory( value )
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder="Search gifs"
                value={ inputValue }
                onChange={ onInputChanged }
            />
        </form>
    )
}
