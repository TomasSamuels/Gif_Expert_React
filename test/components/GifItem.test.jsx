import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItem Test', () => { 

    const title = 'Test Title';
    const url = 'https://test.com/test.jpg'

    test('should match with the snapshot', () => { 
        const { container } = render(<GifItem title={ title }  url={ url } />);
        expect( container ).toMatchSnapshot();
     });

     test('should show the image with the URL and the ALT indicated', () => { 
        render(<GifItem title={ title }  url={ url } />);
        const { src, alt  } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
      });

      test('should show the title', () => { 
        render(<GifItem title={ title }  url={ url } />);
        expect( screen.getByText(title) ).toBeTruthy();
       });

 });
