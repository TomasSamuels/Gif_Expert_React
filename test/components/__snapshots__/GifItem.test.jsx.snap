// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Test should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Title"
      src="https://test.com/test.jpg"
    />
    <p>
      Test Title
    </p>
  </div>
</div>
`;
